lib/web_catalog/object_graph_importer.es6.js

```diff
@@ -134,31 +134,59 @@ foam.CLASS({
           if (stat.isFile()) {
             if (!ogFileRegExp.test(objectGraphFiles[i])) continue;
 
-            // Drop slice(0, -5): ".json".
-            let releaseInfo = objectGraphFiles[i].slice(0, -5).split('_');
-            let bName = releaseInfo[1];
-            let bVersion = releaseInfo[2];
-            let osName = releaseInfo[3];
-            let osVersion = releaseInfo[4];
+            let releaseInfo = this.getReleaseInfo(objectGraphFiles[i]);
 
             this.info(`Extracting APIs from
-                ${bName} ${bVersion} ${osName} ${osVersion}`);
+                ${releaseInfo.browser.name} ${releaseInfo.browser.version}
+                ${releaseInfo.platform.name} ${releaseInfo.platform.version}`);
             promises.push(
                 this.apiExtractorService.extractWebCatalog(filePath)
                 .then(catalog => {
                   this.info(`Importing APIs
-                      ${bName} ${bVersion} ${osName} ${osVersion}`);
-                  return this.apiImporter.import(bName, bVersion, osName,
-                                                 osVersion, catalog);
+                      ${releaseInfo.browser.name} ${releaseInfo.browser.version}
+                      ${releaseInfo.platform.name}
+                      ${releaseInfo.platform.version}`);
+                  return this.apiImporter.import(
+                      releaseInfo.browser.name, releaseInfo.browser.version,
+                      releaseInfo.platform.name, releaseInfo.platform.version,
+                      catalog);
                 }).then(() => {
                   this.info(`APIs imported for
-                                 ${bName} ${bVersion} ${osName} ${osVersion}`);
+                      ${releaseInfo.browser.name} ${releaseInfo.browser.version}
+                      ${releaseInfo.platform.name}
+                      ${releaseInfo.platform.version}`);
                 }));
           }
         }
         return Promise.all(promises);
       },
     },
+    function getReleaseInfo(filename) {
+      const json = require(`${this.objectGraphPath}/${filename}`);
+      if (json.environment && json.environment.browser &&
+          json.environment.browser.name && json.environment.browser.version &&
+          json.environment.platform && json.environment.platform.name &&
+          json.environment.platform.version) {
+        return this.getReleaseInfoFromJSON_(json);
```

Comment:
The if statement has such exhaustive knowledge of what `getReleaseInfoFromJSON_` needs that I think just inlining `json.environment` would be more clear. Or move the conditions into `getReleaseInfoFromJSON_` and throw an exception or return null if it's not met, and handle that here.

Reply:
Good point.